src/__tests__/surveys.test.ts

```diff
@@ -844,6 +843,9 @@ describe('surveys', () => {
     })
 
     describe('branching logic', () => {
+        beforeEach(() => {
+            surveys.getNextSurveyStep = getNextSurveyStep
```

Comment:
i tend to mock the external dependency loader so that more of the flow is being exercised... 

so e.g. in session replay

```
        loadScriptMock.mockImplementation((_ph, _path, callback) => {
            addRRwebToWindow()
            callback()
        })

        assignableWindow.__PosthogExtensions__.loadExternalDependency = loadScriptMock
        ```
        
        otherwise that startup can be broken and tests carry on passing

Reply:
ofc explicit tests somewhere else for it is totally fine too